feature: success message and redirects to list page on successful delete

mianakbarjan opened the pull request:
<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->

## Fixes Issue #8185 

<!-- Remove this section if not applicable -->

Closes #8185 

## Changes proposed
[x] link: After successful delete, redirect back to manage link list page with successfully deleted message
[x] event: After successful delete, redirect back to the manage event list page with a successfully deleted message
[x] milestone: After successful delete, redirect back to the milestone link list page with the successfully deleted message
<!-- List all the proposed changes in your PR -->

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
<!--
[x] - Correct; marked as done
[X] - Correct; marked as done

[ ] - Not correct; marked as **not** done
-->

## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

<!-- Add all the screenshots which support your changes -->

## Note to reviewers

<!-- Add notes to reviewers if applicable -->


<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8591"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

mianakbarjan commented:
Sorry for the unnecessary delete of the branch, I wanted to name it something meaningful, i.e. the issue number but decided to leave it this way and will do it for future PRs.

sital002 reviewed (commented):
The package-lock.json file mustnot be changed unless you changed a dependency in package.json file. I would recommend you to read our local development guide https://linkfree.io/docs/environments/local-development

mianakbarjan commented:
Hi, I've reverted the package-lock.json back to what it is in the current LinkFree repository. I didn't make any changes to it initially and don't why it changed automatically, but I have reverted it back now. Please review.

eddiejaoude commented:
> Hi, I've reverted the package-lock.json back to what it is in the current LinkFree repository. I didn't make any changes to it initially and don't why it changed automatically, but I have reverted it back now. Please review.

It usually happens because people run `npm install` instead of `npm ci` - thanks for reverting it back to `main` 👍

mianakbarjan commented:
> > Hi, I've reverted the package-lock.json back to what it is in the current LinkFree repository. I didn't make any changes to it initially and don't why it changed automatically, but I have reverted it back now. Please review.
> 
> It usually happens because people run `npm install` instead of `npm ci` - thanks for reverting it back to `main` 👍

Oh ok, Thank you for pointing it out. Could you please review the PR and merge it if no other changes need to be made so that I can start working on another issue?

eddiejaoude commented on pages/account/manage/links.js:
```diff
@@ -77,6 +77,12 @@ export default function ManageLinks({ BASE_URL, username, links }) {
+            type="success"
+            message={"Link Deleted Successfully"}
+          />
+        )}
```
I think this could be combined into a single alert using a data object

```js
const alerts = {
   delete: {
      type: "success",
      message: "Link deleted successfully"
   }
   // ...
}
```

eddiejaoude commented on pages/account/manage/events.js:
```diff
@@ -38,9 +38,20 @@ export async function getServerSideProps(context) {
 export default function ManageEvents({ events }) {
   const router = useRouter();
-  const { success } = router.query;
+  const { success, deleted } = router.query;
```
Oh you made the change already, I was still reviewing 😂 

Thank you for the contribution and making the change. I was going to add I think it would be good to also change this line to be `alert` as I think it will simplify lines `64-69`

eddiejaoude reviewed (approved):
Thank you for your contribution 👍 

I will make tweaks via a temporary branch

mianakbarjan commented:
You're welcome!
I was going to reply to your comment asking for more details related to the suggestions you made and what I got from that, but I'll now start working on another issue.

eddiejaoude commented:
Here are the changes I made https://github.com/EddieHubCommunity/LinkFree/pull/8645

mianakbarjan commented:
Got you! This is a much smarter way of doing this 👏
